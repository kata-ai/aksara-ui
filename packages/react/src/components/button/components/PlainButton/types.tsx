import { ButtonBaseProps } from '../Button/types';

export type PlainButtonVariants = 'primary' | 'destructive' | 'disclosure';
export type PlainButtonSizes = 'sm' | 'md';

export interface PlainButtonBaseProps extends Omit<ButtonBaseProps, 'size' | 'block' | 'variant'> {
  variant?: PlainButtonVariants;
  /** True if the button is disabled due to loading */
  isLoading?: boolean;
}
