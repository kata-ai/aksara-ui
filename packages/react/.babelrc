{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": ["last 2 chrome versions", "last 2 firefox versions"]
        }
      }
    ],
    "@babel/preset-react",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "babel-plugin-styled-components"
  ],
  "env": {
    "test": {
      "plugins": ["@babel/plugin-transform-modules-commonjs"]
    }
  }
}
