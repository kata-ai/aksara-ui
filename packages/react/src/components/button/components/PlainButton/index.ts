export { default as PlainButton } from './PlainButton';
export { default as PlainButtonAnchor } from './PlainButtonAnchor';
export * from './PlainButton';
export * from './PlainButtonAnchor';

export * from './types';
