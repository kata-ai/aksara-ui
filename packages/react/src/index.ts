export * from './aksara-provider';
export * from './components';
export * from './layout';
export * from './theme';
export * from './typography';
export * from '@aksara-ui/helpers';
