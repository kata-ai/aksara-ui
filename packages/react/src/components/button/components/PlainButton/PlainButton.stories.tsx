import { action } from '@storybook/addon-actions';
import { Story } from '@storybook/react';
import * as React from 'react';

import PlainButton, { PlainButtonProps } from './PlainButton';

export default {
  title: 'Core/Components/Button/PlainButton',
  component: PlainButton,
  argTypes: {
    variant: {
      options: ['primary', 'destructive'],
    },
    size: {
      options: ['sm', 'md'],
    },
    isLoading: {
      control: 'boolean',
    },
    onClick: {
      action: 'clicked',
    },
  },
};

export const Example: Story<PlainButtonProps> = ({ variant, size, isLoading, onClick }) => (
  <PlainButton type="button" variant={variant} size={size} isLoading={isLoading} onClick={onClick}>
    Push Me
  </PlainButton>
);
Example.args = {
  variant: 'primary',
  size: 'sm',
  isLoading: false,
};
