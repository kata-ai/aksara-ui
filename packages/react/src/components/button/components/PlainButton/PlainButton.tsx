import * as React from 'react';

import { useComponentStyles } from '../../../../system';
import { renderButtonChildren, renderButtonIcon } from '../Button/utils';
import { PlainButtonBaseProps, PlainButtonSizes } from './types';
import { UnstyledButton } from '../UnstyledButton';

export interface PlainButtonProps extends PlainButtonBaseProps, React.ButtonHTMLAttributes<HTMLButtonElement> {
  /** Additional CSS classes to give to the component */
  className?: string;
  /** Additional CSS styles to give to the component */
  style?: React.CSSProperties;
  /** The size of the button. */
  size?: PlainButtonSizes;
  /** True if the button is disabled due to loading */
  isLoading?: boolean;
  /** indicate button is selected */
  selected?: boolean;
}

const PlainButton = React.forwardRef<HTMLButtonElement, PlainButtonProps>(
  ({ children, size = 'sm', icon, iconPosition, disabled, isLoading, variant = 'primary', selected, ...rest }, ref) => {
    const plainButtonStyles = useComponentStyles('plainButton', { size, variant, selected });

    return (
      <UnstyledButton
        disabled={disabled || isLoading}
        ref={ref}
        display="inline-flex"
        flexDirection={iconPosition === 'left' ? 'row' : 'row-reverse'}
        alignItems="center"
        justifyContent="center"
        sx={plainButtonStyles}
        {...rest}
      >
        {renderButtonIcon({ icon, iconPosition, size, isLoading })}
        {renderButtonChildren({ isLoading, variant, children })}
      </UnstyledButton>
    );
  }
);

PlainButton.displayName = 'LinkButton';

/** @deprecated - use `LinkButton` instead */
export const LinkButton = PlainButton;

export default PlainButton;
