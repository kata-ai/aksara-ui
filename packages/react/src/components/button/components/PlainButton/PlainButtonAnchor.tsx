import * as React from 'react';
import { useComponentStyles } from '../../../../system';
import { UnstyledAnchor } from '../../../../typography';
import { renderButtonChildren, renderButtonIcon } from '../Button/utils';
import { PlainButtonBaseProps, PlainButtonSizes } from './types';

export interface PlainButtonAnchorProps extends PlainButtonBaseProps, React.AnchorHTMLAttributes<HTMLAnchorElement> {
  /** Additional CSS classes to give to the component */
  className?: string;
  /** Additional CSS styles to give to the component */
  style?: React.CSSProperties;
  /** The size of the button. */
  size?: PlainButtonSizes;
  /** True if the button is disabled due to loading */
  isLoading?: boolean;
}

const PlainButtonAnchor = React.forwardRef<HTMLAnchorElement, PlainButtonAnchorProps>(
  ({ children, size = 'sm', icon, iconPosition, isLoading, variant = 'primary', ...rest }, ref) => {
    const plainButtonStyles = useComponentStyles('plainButton', { size, variant });

    return (
      <UnstyledAnchor
        ref={ref}
        display="inline-flex"
        flexDirection={iconPosition === 'left' ? 'row' : 'row-reverse'}
        alignItems="center"
        justifyContent="center"
        sx={plainButtonStyles}
        {...rest}
      >
        {renderButtonIcon({ icon, iconPosition, size, isLoading })}
        {renderButtonChildren({ isLoading, variant, children })}
      </UnstyledAnchor>
    );
  }
);

PlainButtonAnchor.displayName = 'PlainButtonAnchor';

export default PlainButtonAnchor;
