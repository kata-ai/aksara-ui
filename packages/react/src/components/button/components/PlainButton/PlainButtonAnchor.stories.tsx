import { IconPen } from '@aksara-ui/icons';
import { Story } from '@storybook/react';
import * as React from 'react';
import PlainButtonAnchor, { PlainButtonAnchorProps } from './PlainButtonAnchor';

export default {
  title: 'Core/Components/Button/PlainButtonAnchor',
  component: PlainButtonAnchor,
  argTypes: {
    variant: {
      options: ['primary', 'destructive', 'disclosure'],
    },
    size: {
      options: ['sm', 'md'],
    },
    isLoading: {
      control: 'boolean',
    },
    onClick: {
      action: 'clicked',
    },
  },
};

const Template: Story<PlainButtonAnchorProps> = ({ variant, size, isLoading, icon, iconPosition, href, target }) => (
  <PlainButtonAnchor
    type="button"
    variant={variant}
    size={size}
    isLoading={isLoading}
    icon={icon}
    iconPosition={iconPosition}
    href={href}
    target={target}
  >
    Push Me
  </PlainButtonAnchor>
);

export const Example: Story<PlainButtonAnchorProps> = Template.bind({});
Example.args = {
  variant: 'primary',
  size: 'md',
  href: 'https://www.youtube.com/watch?v=fdixQDPA2h0',
  target: '_blank',
};

export const WithIcon: Story<PlainButtonAnchorProps> = Template.bind({});
WithIcon.args = {
  ...Example.args,
  icon: IconPen,
  iconPosition: 'left',
};
