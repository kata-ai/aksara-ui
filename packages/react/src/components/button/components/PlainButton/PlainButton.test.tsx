import * as React from 'react';
import { render } from '@testing-library/react';
import { IconDocAdd } from '@aksara-ui/icons';

import PlainButton from './PlainButton';

describe('components/Button', () => {
  describe('<LinkButton />', () => {
    test('renders correctly', () => {
      const { container } = render(<PlainButton>example</PlainButton>);

      expect(container.firstChild).toBeInTheDocument();
    });

    test('renders colors correctly by variant', () => {
      const { container } = render(<PlainButton variant="primary">test button</PlainButton>);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('renders icons correctly', () => {
      const { container } = render(<PlainButton icon={IconDocAdd}>test button</PlainButton>);
      const icon = container.querySelector('svg');

      expect(icon).toBeInTheDocument();
    });
  });
});
