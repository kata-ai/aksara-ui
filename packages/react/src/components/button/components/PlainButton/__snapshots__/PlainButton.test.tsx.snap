// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button <LinkButton /> renders colors correctly by variant 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<button
  class="c0"
>
  <span
    class="Text-sc-1e9atua-0"
  >
    test button
  </span>
</button>
`;
